raft/raft.go

```diff
@@ -390,11 +400,15 @@ func (r *raft) bcastAppend() {
 
 // bcastHeartbeat sends RPC, without entries to all the peers.
 func (r *raft) bcastHeartbeat() {
+	r.bcastHeartbeatWithCtx(nil)
```

Comment:
If there are any pending reads in `readOnly`, we should send the context of the last one with each heartbeat until we get a response. Otherwise if the single MsgHeartbeat is dropped, the read won't be able to finish until the *next* `ReadIndex` succeeds. 

I was originally going to ask why this reused MsgHeartbeat instead of a new message type, but retrying the last context on each heartbeat seems like a good reason.